### PR TITLE
Add shadow edit API

### PR DESCRIPTION
**Pull request by JiuqingSong**
In Shadow Edit mode, all format change will finally be ignored. This can be used for building a live preview feature for format button, to allow user see format result without really apply it.

### CODE REVIEW

File: packages/roosterjs-editor-core/lib/coreApi/getContent.ts [hunk 1]
```diff
@@ -24,15 +25,22 @@ export const getContent: GetContent = (core: EditorCore, mode: GetContentMode):
...
+        content = getTextContent(root);
     } else if (triggerExtractContentEvent || core.lifecycle.isDarkMode) {
-        const clonedRoot = core.contentDiv.cloneNode(true /*deep*/) as HTMLElement;
+        const clonedRoot = cloneNode(root);
         const originalRange = core.api.getSelectionRange(core, true /*tryGetFromCache*/);
-        const path =
-            includeSelectionMarker &&
-            originalRange &&
-            getSelectionPath(core.contentDiv, originalRange);
+        const path = !includeSelectionMarker
```
Comment (flyingbee2012) [inline packages/roosterjs-editor-core/lib/coreApi/getContent.ts:37]: Can we wrap these into a function?

File: demo/scripts/controls/ribbon/RibbonButton.tsx [hunk 2]
```diff
@@ -97,36 +103,65 @@ export default class RibbonButton extends React.Component<RibbonButtonProps, Rib
...
+        buttonOnClick?.(editor, itemName);
+    }, []);
+
+    return (
+        <div
+            key={props.itemName}
+            onClick={onClick}
+            onMouseOver={onMouseOver}
+            className={styles.dropDownItem}>
+            {displayName}
```
Comment (flyingbee2012) [inline demo/scripts/controls/ribbon/RibbonButton.tsx:164]: do we need this localized and add aria label?

File: packages/roosterjs-editor-dom/lib/utils/getInnerHTML.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+/**
+ * Get innerHTML of the given node
+ * @param node The DOM node to get innerHTML from
+ */
+export default function getInnerHTML(node: HTMLElement | DocumentFragment) {
+    if (safeInstanceOf(node, 'HTMLElement')) {
+        return node.innerHTML;
+    } else if (node) {
+        const tempNode = node.ownerDocument.createElement('span');
```
[the remark this answers is not in the document]
Comment (JiuqingSong) [reply packages/roosterjs-editor-dom/lib/utils/getInnerHTML.ts:11]: DocumentFragment doesn't support innerHTML, insert in into another element to get innerHTML

File: packages/roosterjs-editor-core/lib/coreApi/getContent.ts (hunk 1 above)
Comment (JiuqingSong) [reply packages/roosterjs-editor-core/lib/coreApi/getContent.ts:37]: I feel it is already clear enough to directly put it here.

File: demo/scripts/controls/ribbon/RibbonButton.tsx (hunk 2 above)
Comment (JiuqingSong) [reply demo/scripts/controls/ribbon/RibbonButton.tsx:164]: No, this is a demo only.

**Review by flyingbee2012 — APPROVED** [review]

**Pull request merged**